File: dsa-course/how_to_solve/problem-1/ans.py
array1 = ['a', 'b', 'c', 'x']
array2 = ['z', 'y', 'x']

def find_common_item(arr1,arr2):
    commonItem = []
    for i in arr1:  #O(n)
        for j in arr2:   #O(m)
            if i == j:
                commonItem.append(i)
    
    if commonItem:
        print(True)
    else:
        print(False)
# O(n*m)
find_common_item(array1, array2)


# 2nd approch
def find_common_item_2(arr1,arr2):
    commonItem = []
    for i in arr1:
        if i in arr2:
            commonItem.append(i)
    print(commonItem)
    if commonItem:
        print(True)
    else:
        print(False)
# O(n*m)
find_common_item_2(array1, array2)


# 3rd approch

def find_common_item_3(arr1,arr2):
    commonItem = dict()
    for i in arr1:
        commonItem[i] = True
    
    print(commonItem)
    for j in arr2:
        if j in commonItem:
            return True
    return False
    
# O(n*m)
find_common_item_3(array1, array2)

File: dsa-course/how_to_solve/problem-1/ans.js
const array1 = ['a', 'b', 'c', 'x'];
const array2 = ['z', 'y', 'x'];

const findCommon = (arr1,arr2) => {
    let map = {}
    for (let i of arr1) {
        map[i] = true
    }

    console.log(map)
    for (let j of arr2) {
        if (map[j]) {
            return true
        }


    }
    return false
}

console.log(findCommon(array1, array2))


// find common 2

const findCommon2 = (arr1, arr2) => {
    return arr1.some(item => arr2.includes(item))
} 

console.log(findCommon2(array1,array2))